### PR TITLE
CU- d70mwa Unify `bind()` syntax

### PR DESCRIPTION
**Pull request by nomisRev**
This PR removes the newly added `invoke` (before it was released), and deprecates `not` alongside the deprecation of `component1` to unify the computations block API with a single API.

A poll was taken on [KotlinLang](https://kotlinlang.slack.com/archives/C5UPMM0A0/p1611746550025100) to decide between APIs. `bind` was chosen with 62% of the vote.
Alternatives could be implemented locally in a project by adding an extension on the DSL receiver.

### CODE REVIEW

**Review by i-walker — APPROVED** [review]
Thanks to both @nomisRev and @raulraja for the initial pool!

File: arrow-docs/docs/patterns/glossary/README.md [hunk 1]
```diff
@@ -133,19 +133,19 @@ When creating an instance with the `@extension` annotation, the processor genera
 
 ```kotlin:ank:silent
 import arrow.core.Option
-import arrow.core.extensions.fx
+import arrow.core.computations.option
 
-Option.fx {
-  val (a) = Option(1)
-  val (b) = Option(a + 1)
+option.eager<Int> {
```
Comment (aballano) [inline arrow-docs/docs/patterns/glossary/README.md:138]: do we need to specify the type here? I'm wondering since it wasn't needed before, no?

**Review by aballano — APPROVED** [review]
Just some nits

File: arrow-core-data/src/main/kotlin/arrow/core/computations/nullable.kt [hunk 2]
```diff
@@ -1,17 +1,15 @@
...
+    this ?: control().shift(null)
 
-  @Deprecated("The monadic operator for the Arrow 1.x series will become invoke in 0.13", ReplaceWith("()"))
-  suspend fun <B> B?.bind(): B = this()
-
-  @Deprecated("The monadic operator for the Arrow 1.x series will become invoke in 0.13", ReplaceWith("()"))
-  suspend operator fun <B> B?.not(): B = this()
-
-  suspend operator fun <B> B?.invoke(): B = this ?: control().shift(null)
+  suspend fun <B> Option<B>.bind(): B =
```
[the remark this answers is not in the document]
Comment (nomisRev) [reply arrow-core-data/src/main/kotlin/arrow/core/computations/nullable.kt:11]: Yes, exactly. Similar to how you can now `bind` `Validated` inside `either { }`

File: arrow-docs/docs/patterns/glossary/README.md (hunk 1 above)
Comment (nomisRev) [reply arrow-docs/docs/patterns/glossary/README.md:138]: Yes, I had to specify it now or Ank was failing. I'm not sure how it's different from before tho... 😕

**Review by franciscodr — APPROVED** [review]
Thanks @nomisRev!

**Pull request merged**